### ISSUE TITLE
CallBase cannot be used with Delegate mocks

### ISSUE THREAD

**Issue by Veikedo** [AutoFixture/AutoFixture#1083]
First of all, I don't call `CallBase` method explicitly. 

I have this code
```c#
    [Theory]
    [AutoMoqData]
    internal async Task GetAutomationStatistics_should_return_correct_statistics(
      [Frozen] Mock<GetAutomationsLogTotalCount> getAutomationsLogTotalCount,
      [Frozen] GetCurrentlyRunningSchedulesResult getCurrentlyRunningSchedulesResult,
      [Frozen] GetAutomationsTotalCountResult getAutomationsTotalCountResult,
      GetAutomationStatisticsQuery query,
      GetAutomationStatisticsHandler sut,
      CancellationToken ct
    )
    {
      getAutomationsLogTotalCount
        .Setup(_ => _(It.IsAny<GetAutomationsLogTotalCountQuery>(), ct))
        .ReturnsAsync((GetAutomationsLogTotalCountQuery x, CancellationToken _) =>
          new GetAutomationsLogTotalCountResult
          {
            TotalCount = (long) x.LogLevel.GetValueOrDefault()
          }
        );

      var expectedResult = new GetAutomationStatisticsResult
      {
        AutomationsRunningNow = await getCurrentlyRunningSchedulesResult.Items.LongCount(ct),
        TotalAutomations = getAutomationsTotalCountResult.TotalCount,
        ErrorsReported = (long) EapLogLevel.Error,
        WarningsReported = (long) EapLogLevel.Warn,
      };

      var result = await sut.Handle(query, ct);

      result.Should().BeEquivalentTo(expectedResult);
    }
```
Where `GetAutomationsLogTotalCount` is a delegate. And I got the exception

```
AutoFixture.ObjectCreationExceptionWithPath : AutoFixture was unable to create an instance from Moq.Mock`1[AutomationsWeb.Features.Statistics.GetAutomationsLogTotalCount] because creation unexpectedly failed with exception. Please refer to the inner exception to investigate the root cause of the failure.

Request path:
	Moq.Mock`1[AutomationsWeb.Features.Statistics.GetAutomationsLogTotalCount] getAutomationsLogTotalCount
	  Moq.Mock`1[AutomationsWeb.Features.Statistics.GetAutomationsLogTotalCount]

Inner exception messages:
	System.NotSupportedException: CallBase cannot be used with Delegate mocks.
```

_Originally posted by @Veikedo in https://github.com/moq/moq4/issues/706#issuecomment-444523679_

**Comment by zvirja** [comment]
Thanks for firing the issue!

Could you please provide me with a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that helps to reproduce the scenario? It will allow me to copy-paste the code and troubleshoot it much quicker by focusing on the scenario only!

**Comment by zvirja** [comment]
Must be already fixed by #1095
